gnovm/cmd/gno/test.go

```diff
@@ -145,75 +150,72 @@ func execTest(cfg *testCfg, args []string, io *commands.IO) error {
 		}()
 	}
 
+	subPkgs, err := gnomod.SubPkgsFromPaths(paths)
+	if err != nil {
+		return fmt.Errorf("list sub packages: %w", err)
+	}
+
 	buildErrCount := 0
 	testErrCount := 0
-	for _, pkgPath := range pkgPaths {
+	for _, pkg := range subPkgs {
 		if cfg.precompile {
 			if verbose {
-				io.ErrPrintfln("=== PREC  %s", pkgPath)
+				io.ErrPrintfln("=== PREC  %s", pkg.Dir)
 			}
 			precompileOpts := newPrecompileOptions(&precompileCfg{
 				output: tempdirRoot,
 			})
-			err := precompilePkg(importPath(pkgPath), precompileOpts)
+			err := precompilePkg(importPath(pkg.Dir), precompileOpts)
```

Comment:
ditto. importPath should take pkg.Path.
we should use the type system to help here.
notice that in values.go, type PackageValue has Path as a Name type.
a package path is a Name (it is a name in its function).
a package path is a string (like all file paths and dir paths are strings).

Reply:
ok I see that importPath is declared as a `type importPath string`.
this is confusing to me.
something is either a package path, in which case we can just use Name.
we could say `type packagePath Name` but I don't think it's necessary. 

there can be some function that converts a package path to a file path,
but that function would need more context, more than just 1 argument.
it could be a method attached to a structure that holds more context.

Other than such a function, we shouldn't be converting package path Names to file paths or dir paths with type conversions, because nobody should be doing such a thing without calling the official function for such conversions (above).

An "import path" I don't see why we need to create such a name, because it is nothing more than package paths. So ImportPkgPaths is less confusing. Or just Imports of Names.

Any path that is not a package path, should be clearly named as such, by calling it a "file path" or a "dir path" that is a string, never a Name type. Anything like package path, pkgPath, pkg.Path, are all package paths. If you want to associate a file path or a dir path it will have to be called pkg.FilePath or pkg.DirPath.

Reply:
> original var pkgPath also contains Dir

`importPath` is confusing. need to modify precompile related code for that. Will fix in another PR